examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,407 @@
+    total_cells = X.shape[0] * X.shape[1]
+    num_missing_cells = int(total_cells * missing_fraction)
+    row_indices = rng.choice(X.shape[0], num_missing_cells, replace=True)
+    col_indices = rng.choice(X.shape[1], num_missing_cells, replace=True)
+    X_missing = X.copy()
+    X_missing.iloc[row_indices, col_indices] = np.nan
+    return X_missing
+
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y_test.values[first_week], label="Actual transfer")
+
+for missing_fraction in missing_fraction_list:
+    X_train_missing = generate_missing_values(X_train, missing_fraction)
+    X_test_missing = generate_missing_values(X_test, missing_fraction)
+    hgbt.fit(X_train_missing, y_train)
+    y_pred = hgbt.predict(X_test_missing[first_week])
+    rmse = root_mean_squared_error(y_test[first_week], y_pred)
+    ax.plot(
+        y_pred[first_week],
+        label=f"missing_fraction={missing_fraction}, RMSE={rmse:.2f}",
+        alpha=0.5,
+    )
+ax.set(
+    title="Daily energy transfer predictions on data with MCAR values",
+    xticks=[(i + 0.2) * 48 for i in range(7)],
+    xticklabels=["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(loc="lower right")
+
+# %%
+# As expected, the model degrades as the proportion of missing values increases.
+#
+# Support for quantile loss
+# =========================
+#
+# The quantile loss in regression enables a view of the variability or
+# uncertainty of the target variable. For instance, predicting the 5th and 95th
+# percentiles can provide a 90% prediction interval, i.e. the range within which
+# we expect the true value to fall with 90% probability.
+
+from sklearn.metrics import mean_pinball_loss
+
+quantiles = [0.95, 0.05]
+predictions = []
+
+fig, ax = plt.subplots(figsize=(12, 6))
+ax.plot(y_test.values[first_week], label="Actual transfer")
+
+for quantile in quantiles:
+    hgbt_quantile = HistGradientBoostingRegressor(
+        loss="quantile", quantile=quantile, **common_params
+    )
+    hgbt_quantile.fit(X_train, y_train)
+    y_pred = hgbt_quantile.predict(X_test[first_week])
+
+    predictions.append(y_pred)
+    score = mean_pinball_loss(y_test[first_week], y_pred)
+    ax.plot(
+        y_pred[first_week],
+        label=f"quantile={quantile}, pinball loss={score:.2f}",
+        alpha=0.5,
+    )
+
+ax.fill_between(
+    range(len(predictions[0][first_week])),
+    predictions[0][first_week],
+    predictions[1][first_week],
+    color=colors[0],
+    alpha=0.1,
+)
+ax.set(
+    title="Daily energy transfer predictions with quantile loss",
+    xticks=[(i + 0.2) * 48 for i in range(7)],
+    xticklabels=["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"],
+    xlabel="Time of the week",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(loc="lower right")
+
+# %%
+# We observe a tendence to over-estimate the energy transfer. This could be be
```

Comment:
This can already be observed with the mean regression (squared error). It might give us a hint that we should search for better hyperparameters.

Reply:
A bit lower in the text we mention:
> One can still improve the quality of such estimations by:
> - better tuning of the model hyperparameters

Do you think we should explicitly tune them? Or behind the scenes and just report the results of a grid search?